DisCatSharp/Entities/Message/DiscordMessageBuilder.cs

```diff
@@ -326,37 +264,37 @@ public DiscordMessageBuilder AddEmbed(DiscordEmbed embed)
 	/// <returns>The current builder to be chained.</returns>
 	public DiscordMessageBuilder AddEmbeds(IEnumerable<DiscordEmbed> embeds)
 	{
-		this._embeds.AddRange(embeds);
+		this.EmbedsInternal.AddRange(embeds);
 		return this;
 	}
 
+	[Deprecated("Replaced by AddMention")]
+	public DiscordMessageBuilder WithAllowedMention(IMention allowedMention)
+		=> this.AddMention(allowedMention);
+
 	/// <summary>
 	///     Sets if the message has allowed mentions.
 	/// </summary>
-	/// <param name="allowedMention">The allowed Mention that should be sent.</param>
+	/// <param name="mention">The allowed Mention that should be sent.</param>
 	/// <returns>The current builder to be chained.</returns>
-	public DiscordMessageBuilder WithAllowedMention(IMention allowedMention)
+	public DiscordMessageBuilder AddMention(IMention mention)
 	{
-		if (this.Mentions != null)
-			this.Mentions.Add(allowedMention);
-		else
-			this.Mentions = [allowedMention];
-
+		this.MentionsInternal.Add(mention);
 		return this;
 	}
 
+	[Deprecated("Replaced by AddMentions")]
+	public DiscordMessageBuilder WithAllowedMentions(IEnumerable<IMention> allowedMentions)
+		=> this.AddMentions(allowedMentions);
+
 	/// <summary>
 	///     Sets if the message has allowed mentions.
 	/// </summary>
-	/// <param name="allowedMentions">The allowed Mentions that should be sent.</param>
+	/// <param name="mentions">The allowed Mentions that should be sent.</param>
 	/// <returns>The current builder to be chained.</returns>
-	public DiscordMessageBuilder WithAllowedMentions(IEnumerable<IMention> allowedMentions)
+	public DiscordMessageBuilder AddMentions(IEnumerable<IMention> mentions)
 	{
-		if (this.Mentions != null)
-			this.Mentions.AddRange(allowedMentions);
-		else
-			this.Mentions = allowedMentions.ToList();
-
+		this.MentionsInternal.AddRange(mentions);
 		return this;
 	}
```

Comment:
Still thinking about whether to keep WithAllowedMention[s], which would mean to change all other builders, or to use AddMention[s]

Reply:
@byterand what feels more natural to you?

Reply:
staying with WithAllowedMention / WithAllowedMentions